src/test/java/com/puppycrawl/tools/checkstyle/XDocsPagesTest.java

```diff
@@ -244,4 +267,230 @@ private static void testCheckstyleXml(String fileName, String code, String unser
+                final Node section = sources.item(position);
+                final String sectionName = section.getAttributes().getNamedItem("name")
+                        .getNodeValue();
+
+                if ("Content".equals(sectionName) || "Overview".equals(sectionName)) {
+                    continue;
+                }
+
+                System.out.println(fileName + " section '" + sectionName + "'");
+
+                Assert.assertTrue(fileName + " section '" + sectionName
+                        + "' shouldn't end with 'Check'", !sectionName.endsWith("Check"));
+
+                Object clss = null;
+
+                try {
+                    clss = moduleFactory.createModule(sectionName);
+                }
+                catch (CheckstyleException e) {
+                    Assert.fail(fileName + " couldn't find class: " + sectionName);
+                }
+
+                int subSectionPos = 0;
+                for (Node subSection : getChildrenElements(section)) {
+                    final String subSectionName = subSection.getAttributes().getNamedItem("name")
+                            .getNodeValue();
+
+                    // can be in different orders, and completely optional
+                    if (subSectionName.equals("Notes"))
+                        continue;
+                    if (subSectionName.equals("Rule Description"))
+                        continue;
+                    // TODO: merge this with description
+                    if (subSectionName.equals("Use Case"))
+                        continue;
+
+                    if (subSectionPos == 1 && !subSectionName.equals("Properties")) {
+                        validatePropertySection(fileName, sectionName, null, clss);
+                        subSectionPos++;
+                    }
+
+                    Assert.assertEquals(fileName + " section '" + sectionName
+                            + "' should be in order", getSubSectionName(subSectionPos),
+                            subSectionName);
+
+                    switch (subSectionPos) {
+                    case 0:
+                        // TODO: should we enforce no <source>?
+                        break;
+                    case 1:
+                        validatePropertySection(fileName, sectionName, subSection, clss);
+                        break;
+                    case 2:
+                        break;
+                    case 3:
+                        break;
+                    case 4:
+                        validatePackageSection(fileName, sectionName, subSection, clss);
+                        break;
+                    case 5:
+                        validateParentSection(fileName, sectionName, subSection);
+                        break;
+                    }
+
+                    subSectionPos++;
+                }
+            }
+        }
+    }
+
+    private static Object getSubSectionName(int subSectionPos) {
+        switch (subSectionPos) {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Switch statements should have a default label](https://www.codacy.com/app/checkstyle/checkstyle/file/1183653445/issues/source?bid=2193633&fileBranchId=2921952#l361)